src/cunumeric/index/wrap.cc

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/wrap.h"
+#include "cunumeric/index/wrap_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <int DIM>
+struct WrapImplBody<VariantKind::CPU, DIM> {
+  void operator()(const AccessorWO<Point<DIM>, 1>& out,
+                  const Pitches<0>& pitches_out,
+                  const Rect<1>& out_rect,
+                  const Pitches<DIM - 1>& pitches_in,
+                  const Rect<DIM>& in_rect,
+                  const bool dense) const
+  {
+    const size_t start     = out_rect.lo[0];
+    size_t end             = out_rect.hi[0];
+    const size_t in_volume = in_rect.volume();
+    size_t out_idx         = 0;
+    if (dense) {
+      auto outptr = out.ptr(out_rect);
+      for (size_t i = start; i <= end; i++) {
+        const size_t input_idx = i % in_volume;
+        auto point             = pitches_in.unflatten(input_idx, in_rect.lo);
+        outptr[out_idx]        = point;
+        out_idx++;
+      }
+    } else {
+      for (size_t i = start; i <= end; i++) {
+        const size_t input_idx = i % in_volume;
+        auto point             = pitches_in.unflatten(input_idx, in_rect.lo);
+        auto point_out         = pitches_out.unflatten(out_idx, out_rect.lo);
+        out[point_out]         = point;
+        out_idx++;
```

Comment:
You don't need to fiddle with `out_idx` in this case

Reply:
Actually, you still need to use `out_idx` if you want to unflatten, but since `i` ranges from `out_rect.lo[0]` to `out_rect.ho[0]` already, could you just do `out[i] = point;`?

Reply:
that it right. fixing this